Add Kubernetes on AWS EKS Hello World Tutorial

metral opened the pull request.

metral commented:
The corresponding code for the example linked in the tutorial: https://github.com/pulumi/pulumi-eks/pull/61

PTAL at both as the linked PR is a dependency of this tutorial, and needs to be merged first.

joeduffy commented on quickstart/aws/tutorial-eks.md:
````diff
@@ -0,0 +1,303 @@
+1. Add the required dependencies:
+
+	```bash
+	$ yarn add @pulumi/eks @pulumi/aws-infra
````
I _think_ we tend to prefer `npm` in these tutorials. So

```
$ npm install --save @pulumi/eks @pulumi/aws-infra
```

joeduffy commented on quickstart/aws/tutorial-eks.md:
````diff
@@ -0,0 +1,303 @@
+1.  To preview and deploy changes, run `pulumi update`. The command shows a preview of the resources that will be created and prompts on whether to proceed with the deployment.  Note that the stack itself is counted as a resource, though it does not correspond to a physical cloud resource.
+
+    ```bash
+	$ pulumi update
````
Nit, tabs versus spaces?

joeduffy commented on quickstart/aws/tutorial-eks.md:
```diff
@@ -0,0 +1,303 @@
+	Resources:
+		+ 42 to create
+
+	clusterng (eks-demo):
```
Maybe make a note here that you should select "yes" to confirm the deployment. I'd probably just copy whatever we do in other tutorials.

joeduffy commented:
Hmm, having a tutorial in addition to the example is highly valuable ... Why not go forward with it?

metral commented:
I could add the proposed example in that same `pulumi/examples/aws-ts-eks` dir as the existing one, but I wasn't sure if that'd be unnecessarily extraneous, not to mention writing a tutorial for it.

I can still proceed forward, just lmk what makes the most sense.

joeduffy commented:
Having a tutorial on the docs website tends to significantly increase its visibility and utility. I don't think we have one for EKS currently -- it's a shame not to publish the excellent one you've written.

I'm on the fence on the question of the examples repo, however, I do think showing how to deploy K8s objects into a newly provisioned cluster is sufficiently subtle and interesting that it's worth having.

lukehoban commented:
Yes - let's definitely add this tutorial to the docs site.  

And totally fine having both `aws-ts-eks` and `aws-ts-eks-hello-world` in the examples repo.

metral commented on quickstart/aws/tutorial-eks.md:
```diff
@@ -0,0 +1,303 @@
+
+1.  Run `pulumi update` to preview and deploy the changes. You'll see two changes: the `Namespace` in Kubernetes to deploy into, and the `Deployment` resource for the new NGINX app. Pulumi understands which changes to a given cloud resource can be made in-place, and which require replacement, and computes the minimally disruptive change to achieve the desired state.
+
+    As part of the update, you'll see the two new objects in the output:
```
Great point. I've included the wording @lukehoban[ used in the community slack](https://pulumi-community.slack.com/archives/C84L4E3N1/p1545597765076500?thread_ts=1545559366.075500&cid=C84L4E3N1) for reference.

metral commented:
I've reopened the PR for this EKS tutorial, and addressed all feedback. 

I've also created a new `aws-ts-eks-hello-world` example in `pulumi/examples` that this tutorial depends on for its link to source code that will have to be merged in first: https://github.com/pulumi/examples/pull/219

PTAL
